cocos/core/algorithm/move.ts

```diff
@@ -1,28 +1,32 @@
 import { assertsArrayIndex } from '../data/utils/asserts';
 
 /**
- * Moves an array element to new location.
- * @param array The array.
- * @param index Index to the array element to move.
- * @param newIndex New array index it should have.
+ * @en Circle shift an array once in range [first, last]. If `first < last`, then will do left circle shift;
+ * If `first > last`, then will do right circle shift; If `first == last`, there is not effect.
+ * @zh 在 [first, last] 范围做一次环形移动。如果 `first < last`，进行左环形移动；如果 `first > last`，进行右环
+ * 形移动；如果 `first == last`，将没有效果。
+ * @param array @en The array to be rotated. @zh 被修改的数组。
+ * @param first @en The beginning of the original range. @zh 移动区间的起始坐标。
+ * @param last @en The end of the original range. @zh 移动区间的结束坐标。
+ * @return @en The input array. @zh 输入的数组。
  * @engineInternal
  */
-export function move<T> (array: T[], index: number, newIndex: number) {
-    assertsArrayIndex(array, index);
-    assertsArrayIndex(array, newIndex);
-    if (index === newIndex) {
+export function shift<T> (array: T[], first: number, last: number) {
```

Comment:
Should this file be renamed to `shift.ts` or `array-shift.ts` ?

Reply:
I thought it before. But as i thought, shift can be a special move operation, so i didn't rename it.